security/advancedtls/advancedtls.go

```diff
@@ -79,21 +81,67 @@ type GetRootCAsResults struct {
 	TrustCerts *x509.CertPool
 }
 
-// RootCertificateOptions contains a field and a function for obtaining root
-// trust certificates.
-// It is used by both ClientOptions and ServerOptions.
-// If users want to use default verification, but did not provide a valid
-// RootCertificateOptions, we use the system default trust certificates.
+// RootCertificateOptions contains options to obtain root trust certificates
+// for both the client and the server.
+// At most one of the options could be set. If none of them are set, we
+// use the system default trust certificates.
 type RootCertificateOptions struct {
-	// If field RootCACerts is set, field GetRootCAs will be ignored. RootCACerts
-	// will be used every time when verifying the peer certificates, without
-	// performing root certificate reloading.
+	// If RootCACerts is set, it will be used every time when verifying
+	// the peer certificates, without performing root certificate reloading.
 	RootCACerts *x509.CertPool
-	// If GetRootCAs is set and RootCACerts is nil, GetRootCAs will be invoked
-	// every time asked to check certificates sent from the server when a new
-	// connection is established.
-	// This is known as root CA certificate reloading.
-	GetRootCAs func(params *GetRootCAsParams) (*GetRootCAsResults, error)
+	// If GetRootCertificates is set, it will be invoked to obtain root certs in
```

Comment:
Just curious, is there a reason for not caching the root certificates, rather than calling the callback for each connection?

Reply:
Reloading per connection is more flexible, and users can do whatever they want on top of that. Also, current implemented caching approach is for PEM files. if users don't want to reload from PEM files, they would need to implement their Provider implementations, but if their use cases are simple enough they can just choose callback approach. 
It's not difficult to expose it in Go, and can provide more choices, so I expose it here.